### PR TITLE
docs: update upgrade guide with `talosctl upgrade-k8s`

### LINKED ISSUE #2520: docs: update upgrade docs with `talosctl upgrade-k8s`
## Feature Request

With 0.6.1 we can do automated k8s upgrades

### PR DESCRIPTION
**Pull request by smira**
This offers automated upgrades for Kubernetes control plane.

Fixes #2520

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2526)

### CODE REVIEW

**Comment by smira** [comment]
https://5f61399ef572f6000803637c--talos-dev.netlify.app/docs/v0.6/en/guides/upgrading

File: docs/website/content/v0.6/en/guides/upgrading/index.md [hunk 1]
```diff
@@ -26,10 +26,71 @@ In order to edit the control plane, we will need a working `kubectl` config.
...
 
 ```bash
-talosctl kubeconfig
+talosctl --nodes <master node> kubeconfig
 ```
 
-### API Server
+### Automated Kubernetes Upgrade
+
+New command `talosctl upgrade-k8s` can be used to automate the Kubernetes upgrade process, this command was first released with Talos 0.6.1.
```
Comment (andrewrynhard) [inline docs/website/content/v0.6/en/guides/upgrading/index.md:34]:
```suggestion
In Talos v0.6.1 we introduced the `upgrade-k8s` command in `talosctl`.
This command can be used to automate the Kubernetes upgrade process.
For example, to upgrade from Kubernetes v1.18.6 to v1.19.0 run:
```

**Review by andrewrynhard — DISMISSED** [review]

File: docs/website/content/v0.6/en/guides/upgrading/index.md [hunk 2]
```diff
@@ -26,10 +26,71 @@ In order to edit the control plane, we will need a working `kubectl` config.
...
+
+```bash
+$ talosctl --nodes <master node> upgrade-k8s --from 1.18.6 --to 1.19.0
+updating pod-checkpointer grace period to "0m"
+sleeping 5m0s to let the pod-checkpointer self-scheckpoint be updated
```
Comment (rsmitty) [inline docs/website/content/v0.6/en/guides/upgrading/index.md:39]: Not sure if this is a typo in the actual upgrade code, but `self-scheckpoint`

Comment (rsmitty) [inline docs/website/content/v0.6/en/guides/upgrading/index.md:39]:
```suggestion
sleeping 5m0s to let the pod-checkpointer self-checkpoint be updated
```

File: docs/website/content/v0.6/en/guides/upgrading/index.md [hunk 3]
```diff
@@ -26,10 +26,71 @@ In order to edit the control plane, we will need a working `kubectl` config.
...
+updating daemonset "kube-apiserver" to version "1.19.0"
+updating daemonset "kube-controller-manager" to version "1.19.0"
+updating daemonset "kube-scheduler" to version "1.19.0"
+updating daemonset "kube-proxy" to version "1.19.0"
+updating pod-checkpointer grace period to "5m0s"
+```
+
+### Manual Kubernetes Upgrade
+
+Kubernetes can be upgraded manually as well by following the steps outlined below, they are equivalent to the steps performed by the `talosctl upgrade-k8s` command.
```
Comment (rsmitty) [inline docs/website/content/v0.6/en/guides/upgrading/index.md:50]:
```suggestion
Kubernetes can be upgraded manually as well by following the steps outlined below.
They are equivalent to the steps performed by the `talosctl upgrade-k8s` command.
```

File: docs/website/content/v0.6/en/guides/upgrading/index.md [hunk 4]
```diff
@@ -26,10 +26,71 @@ In order to edit the control plane, we will need a working `kubectl` config.
...
+
+#### pod-checkpointer
+
+Talos runs `pod-checkpointer` component which helps to recover control plane components (specifically, API server) if control plane is not healthy, but the way
+checkpoints interact with API server upgrade might make upgrade take a lot of time due to race condition in API server listen port.
```
Comment (rsmitty) [inline docs/website/content/v0.6/en/guides/upgrading/index.md:55]:
```suggestion
Talos runs `pod-checkpointer` component which helps to recover control plane components (specifically, API server) if control plane is not healthy.
However, the way checkpoints interact with API server upgrade may make an upgrade take a lot longer due to a race condition in API server listen port.
```

**Comment by rsmitty** [comment]
Nice job dude! Fixed a few sentences.

File: docs/website/content/v0.6/en/guides/upgrading/index.md (hunk 1 above)
Comment (smira) [reply docs/website/content/v0.6/en/guides/upgrading/index.md:34]: thanks!

File: docs/website/content/v0.6/en/guides/upgrading/index.md (hunk 2 above)
Comment (smira) [reply docs/website/content/v0.6/en/guides/upgrading/index.md]: :facepalm: that's in the code, I'll fix it as well

File: docs/website/content/v0.6/en/guides/upgrading/index.md (hunk 3 above)
Comment (smira) [reply docs/website/content/v0.6/en/guides/upgrading/index.md]: :+1:

File: docs/website/content/v0.6/en/guides/upgrading/index.md (hunk 4 above)
Comment (smira) [reply docs/website/content/v0.6/en/guides/upgrading/index.md]: :heart:

**Review by andrewrynhard — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**